### PR TITLE
HPCC-23665 Calculate and store workflow cost

### PR DESCRIPTION
**Pull request by shamser**
Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by shamser** [comment]
### **Don't review this is a test**

**Comment by shamser** [comment]
@ghalliday

File: ecl/eclagent/eclagent.hpp [hunk 1]
```diff
@@ -25,6 +25,8 @@ version: "1.0"
     name: myeclagent
     thorConnectTimeout: 600
     traceLevel: 0
+    costs:
```
Comment (ghalliday) [inline ecl/eclagent/eclagent.hpp]: @richardkchapman should we be adding commented out and documented items into values.yaml for each value that makes sense to be overridden?

File: ecl/eclagent/eclagent.hpp [hunk 2]
```diff
@@ -25,6 +25,8 @@ version: "1.0"
...
+      workflowMachine: 0.0
```
Comment (ghalliday) [inline ecl/eclagent/eclagent.hpp]: Should this default to non-zero?

Comment (shamser) [reply ecl/eclagent/eclagent.hpp]: This file was deleted in HPCC-23609 and the yaml definition moved to ecl/agentexec/agentexec.cpp.

File: ecl/eclagent/eclagent.hpp (hunk 1 above)
Comment (shamser) [reply ecl/eclagent/eclagent.hpp]: This file was deleted in HPCC-23609 and the yaml definition moved to ecl/agentexec/agentexec.cpp.

**Comment by shamser** [comment]
@ghalliday

File: ecl/eclagent/eclagent.cpp [hunk 3]
```diff
@@ -563,6 +563,8 @@ EclAgent::EclAgent(IConstWorkUnit *wu, const char *_wuid, bool _checkVersion, bo
             w->setXmlParams(_queryXML);
         updateSuppliedXmlParams(w);
     }
+    IPropertyTree *costs = queryCostsConfiguration();
+    workflowMachineCost = costs ? costs->getPropReal("@workflowMachine", 0.0): 0.0;
```
Comment (ghalliday) [inline ecl/eclagent/eclagent.cpp]: @agent @hthor or @roxie is probably more appropriate.  It isn't just performing workflow - it executes some workflow items.
@richardkchapman what name do you think it should use
e.g.
cost:
   agent: 10

**Comment by shamser** [comment]
@ghalliday

**Pull request merged**